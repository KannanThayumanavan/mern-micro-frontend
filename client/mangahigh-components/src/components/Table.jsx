import React from 'react';
import PropTypes from 'prop-types';

const Table = ({ tableData }) => (
	<table className="table table-striped table-hover table-light">
	  <thead>
	    <tr>
	    	{tableData.header.map((header) => <th key={header} scope="col">{header}</th>)}		      
	    </tr>
	  </thead>
	  <tbody>
	  	{tableData.rows.map((row) => (
	  		<tr key={row}>
	  			{row.map((data, index) => <td key={`${data}_${index}`}>{data}</td>)}
	  		</tr>
	  	))}
	  </tbody>
	</table>
);

Table.propTypes = {
	tableData: PropTypes.shape({
		header: PropTypes.arrayOf(PropTypes.string).isRequired,
		rows: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
	}),
};

export default Table;
