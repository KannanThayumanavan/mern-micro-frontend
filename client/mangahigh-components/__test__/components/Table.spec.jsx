import React from 'react';
import { shallow } from 'enzyme';
import Table from '../../src/components/Table';

describe('Test Table Component', () => {
  const tableData = {
    header: ['Heading 1', 'Heading 2'],
    rows: [
      ['Row1 Data1', 'Row1 data2'],
      ['Row2 Data1', 'Row2 data2'],
    ],
  };
	it('should render correctly with required props', () => {
		const renderedModule = shallow(<Table tableData={tableData} />);
    expect(renderedModule).toMatchSnapshot();
	});
});