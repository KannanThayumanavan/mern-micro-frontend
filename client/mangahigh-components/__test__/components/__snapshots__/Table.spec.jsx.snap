// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Table Component should render correctly with required props 1`] = `
<table
  className="table table-striped table-hover table-light"
>
  <thead>
    <tr>
      <th
        key="Heading 1"
        scope="col"
      >
        Heading 1
      </th>
      <th
        key="Heading 2"
        scope="col"
      >
        Heading 2
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="Row1 Data1,Row1 data2"
    >
      <td
        key="Row1 Data1_0"
      >
        Row1 Data1
      </td>
      <td
        key="Row1 data2_1"
      >
        Row1 data2
      </td>
    </tr>
    <tr
      key="Row2 Data1,Row2 data2"
    >
      <td
        key="Row2 Data1_0"
      >
        Row2 Data1
      </td>
      <td
        key="Row2 data2_1"
      >
        Row2 data2
      </td>
    </tr>
  </tbody>
</table>
`;
